web/pdf_thumbnail_view.js

```diff
@@ -32,35 +32,52 @@ var THUMBNAIL_CANVAS_BORDER_WIDTH = 1; // px
  *   but increases the overall memory usage. The default value is false.
  */
 
+const TempImageFactory = (function TempImageFactoryClosure() {
```

Comment:
Yes, I wanted to keep the actual canvas private, and that cannot be done using classes (at least not as they are specified in ES6, but hopefully that changes in a future EcmaScript version).

Also, please keep in mind that currently all `PDFThumbnailView`s share just *one* temporary canvas, so a class would be overkill here.